areas/mysql/src/AzureMcp.MySql/MySqlSetup.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using AzureMcp.Core.Areas;
+using AzureMcp.Core.Commands;
+using AzureMcp.MySql.Commands.Database;
+using AzureMcp.MySql.Commands.Server;
+using AzureMcp.MySql.Commands.Table;
+using AzureMcp.MySql.Services;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace AzureMcp.MySql;
+
+public class MySqlSetup : IAreaSetup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddSingleton<IMySqlService, MySqlService>();
+    }
+
+    public void RegisterCommands(CommandGroup rootGroup, ILoggerFactory loggerFactory)
+    {
+        var mysql = new CommandGroup("mysql", "MySQL operations - Commands for managing Azure Database for MySQL Flexible Server resources. Includes operations for listing servers and databases, executing SQL queries, managing table schemas, and configuring server parameters.");
+        rootGroup.AddSubGroup(mysql);
+
+        var database = new CommandGroup("database", "MySQL database operations");
+        mysql.AddSubGroup(database);
+        database.AddCommand("list", new DatabaseListCommand(loggerFactory.CreateLogger<DatabaseListCommand>()));
+        database.AddCommand("query", new DatabaseQueryCommand(loggerFactory.CreateLogger<DatabaseQueryCommand>()));
+
+        var table = new CommandGroup("table", "MySQL table operations");
+        mysql.AddSubGroup(table);
+        table.AddCommand("list", new TableListCommand(loggerFactory.CreateLogger<TableListCommand>()));
+        table.AddCommand("schema", new TableSchemaCommand(loggerFactory.CreateLogger<TableSchemaCommand>()));
+
+        var server = new CommandGroup("server", "MySQL server operations");
+        mysql.AddSubGroup(server);
+        server.AddCommand("list", new ServerListCommand(loggerFactory.CreateLogger<ServerListCommand>()));
+        server.AddCommand("config", new ServerConfigCommand(loggerFactory.CreateLogger<ServerConfigCommand>()));
+        server.AddCommand("param", new ServerParamCommand(loggerFactory.CreateLogger<ServerParamCommand>()));
+        server.AddCommand("setparam", new ServerSetParamCommand(loggerFactory.CreateLogger<ServerSetParamCommand>()));
```

Comment:
Can this be param set instead of setparam?

Reply:
We also used param & setparam for postgres server.

If we decide to modify it, let's make sure we also update the postgres one to keep them aligned.

Reply:
Also wanted to ask if a command always has to be one word. Can we use 2 words with space as command name ?

Reply:
either one word, or a hyphenated word.

Reply:
Please use `param set`.  I'll create an issue to update the postgres one.

Reply:
To provide some context, postgres used setparam because hyphenated names were not supported at that time.

Reply:
If we are using ObjectVerb, then should we make it  param get and param set.

Reply:
done https://github.com/Azure/azure-mcp/pull/855/commits/dddf8c3b2967fcd30d65aee29eb4f9eba3bebdea